packages/core/status/core-status-server-internal/src/log_plugins_status.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Observable } from 'rxjs';
+import { pairwise, takeUntil, map, filter, startWith } from 'rxjs/operators';
+import type { PluginName } from '@kbn/core-base-common';
+import { ServiceStatus, ServiceStatusLevels } from '@kbn/core-status-common';
+
+export type ServiceStatusWithName = ServiceStatus & {
+  pluginName: PluginName;
+};
+
+export type PluginStatus = 'unavailable' | 'critical' | 'degraded' | 'available';
+export type PluginsByStatus = Record<PluginStatus, ServiceStatusWithName[]>;
+
+export interface PluginStatusChanges {
+  status: PluginsByStatus;
+  updates: PluginsByStatus;
+  updated: number;
+  total: number;
+}
+
+export const getPluginsStatusChanges = (
+  plugins$: Observable<Record<PluginName, ServiceStatus>>,
+  stop$: Observable<void>
+): Observable<PluginStatusChanges> => {
+  return plugins$.pipe(
+    startWith({}), // consider all plugins unavailable by default
+    takeUntil(stop$),
+    pairwise(),
+    map(([oldStatus, newStatus]) => getPluginsStatusDiff(oldStatus, newStatus)),
+    filter((statusChanges) => statusChanges.updated > 0)
```

Comment:
We only emit events if at least one plugin status has changed.
NB we no longer _debounce_ or _throttleTime_ here,
as we rely on https://github.com/elastic/kibana/pull/128324 improvements to have a lot less emissions.

Reply:
> NB we no longer debounce or throttleTime here

It's great to know we don't need it anymore performance-wise, but I still wonder if keeping debounce/throttling with a small time range wouldn't be better to avoid the edge case of a "blinking" plugin flooding the logs, wdyt?

Reply:
I am not sure `debounce` or `throttleTime` are the best candidates for what we want to achieve:
- `debounce` only emits the most recent event when the time comes, and the others are simply lost.
- `throttleTime` also ignores events for some time after a given emission.

I was thinking of a combination of `map` + `filter` using an external temporary cache to keep track of number of emissions of each _pluginName_, and filter them up to a certain time or number of repetitions, having some sort of factorised message:`Plugin A switched status 38 times the last minute: